### PR TITLE
Rerun go-test and go-test-32bits tests up to 3 times before failing

### PR DESCRIPTION
**Pull request by dhiaayachi**
This PR is to rerun up to 3 times go tests before failing the CI.

### CODE REVIEW

**Review by markan — APPROVED** [review]
Thanks for doing this

File: .circleci/scripts/rerun-fails-report.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (dhiaayachi) [reply .circleci/scripts/rerun-fails-report.sh:1]: I was also thinking of adding metrics collection in an artifactory  file as a second step. To be able to know the top 10 offenders for example.

**Review by dnephin — COMMENTED** [review]
Thanks for working on this!

Are you sure this will work in the OSS repo like it does in Ent?  The `GITHUB_TOKEN` in the script comes from a CircleCI Context, but we can't use that context in OSS because it would expose secrets to all external contributors.

That's why I only added it to enterprise originally and we haven't moved it to OSS.

We can still add the retries, but posting the comments to github won't work, and it will cause the build to fail, which kind of negates the retries.

There's a way to make this work without exposing secrets using https://docs.github.com/en/actions/reference/events-that-trigger-workflows#check_suite

I can share more details with you in slack.

**Comment by dhiaayachi** [comment]
Thank you for taking a look at this @dnephin 
As discussed I changed this to not post comment, on the PR. I left the reporting job and changed it to echo the report instead so that it give the PR author a convenient way to check for test rerun.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/450859.

**Pull request merged**